### PR TITLE
Remove Caseworker and Team Leader assignment from internal contacts

### PR DESCRIPTION
**Pull request by lozette**
Due to confustion from users about what 1) a "team leader" actually is, and 2) the difference between a caseworker and the "assigned to" person on a project we have decided to hide these two fields from users.

The team leader will always be one of two people, and the "assigned to" person is the field we want users to populate & pay attention to.

Along with removing the two fields from the view, we have repurposed some of the tests to test the assignment of "assigned_to" still works correctly.

Additionally, the Caseworker Mailer has been renamed to AssignedToMailer, and a message is now sent to the `assigned_to` person when they are assigned to a project.

## Changes

_Add a summary of the changes in this pull request_

## Checklist

- [x] Attach this pull request to the appropriate card in Trello.
- [x] Update the `CHANGELOG.md` if needed.

### CODE REVIEW

**Comment by lozette** [comment]
[Remove the team lead and caseworker fields from the internal contacts tab](https://trello.com/c/VF5vtYe9/1277-remove-the-team-lead-and-caseworker-fields-from-the-internal-contacts-tab)

File: app/mailers/assigned_to_mailer.rb [hunk 1]
```diff
@@ -0,0 +1,12 @@
+class AssignedToMailer < ApplicationMailer
+  def assigned_notification(user, project)
+    template_mail(
+      "ec6823ec-0aae-439b-b2f9-c626809b7c61",
```
Comment (lozette) [inline app/mailers/assigned_to_mailer.rb:4]: Do we need to amend this template?

File: spec/requests/assignments_controller_spec.rb [hunk 2]
```diff
@@ -91,70 +91,6 @@
...
-      spec.run
-    end
-
-    subject(:perform_request) do
-      post project_assign_caseworker_path(project_id), params: {conversion_project: {caseworker_id: caseworker.id}}
-      response
-    end
-
-    context "when the project has been assigned a caseworker previously" do
-      let(:caseworker_assigned_at) { DateTime.yesterday.at_midday }
```
[the remark this answers is not in the document]
Comment (lozette) [reply spec/requests/assignments_controller_spec.rb:128]: We're going to rename this column in a separate ticket

File: app/mailers/assigned_to_mailer.rb (hunk 1 above)
Comment (beeLorna) [reply app/mailers/assigned_to_mailer.rb:4]: Is this a question regarding the contents of the email template?

Comment (lozette) [reply app/mailers/assigned_to_mailer.rb:4]: Yeah, not sure if it makes reference to Caseworkers specifically! I can ask in the team.

Comment (lozette) [reply app/mailers/assigned_to_mailer.rb:4]: Nope, it doesn't!

**Pull request merged**